frontend/vue/components/CodeExercise/CodeExercise.vue

```diff
@@ -84,16 +95,21 @@ export default defineComponent({
     const initialCodeElement = slotWrapper.getElementsByTagName('pre')[0]
     this.code = initialCodeElement?.textContent?.trim() ?? ''
     this.initialCode = this.code
+    this.id = lastId++
   },
   methods: {
     run () {
       const codeOutput: any = this.$refs.output
+      const pageTitle = document.querySelector('.page-title')?.textContent
```

Comment:
I think that the route name is better since we have several `Introduction` units and it the name will change with the language.